src/node.cc

```diff
@@ -1144,7 +1143,7 @@ extern "C" void node_module_register(void* m) {
     mp->nm_link = modlist_linked;
     modlist_linked = mp;
   } else {
-    modpending = mp;
+    Environment::GetThreadLocalEnv()->modpending = mp;
```

Comment:
Almost any use of `Environment::GetThreadLocalEnv()` is going to be buggy, because Node.js generally seeks to support embedders who create multiple `Environment`s per thread :(

Reply:
@addaleax Thanks, I was unaware of that. I've migrated to using a thread local instead, which should properly support multiple `Environment`s per thread.